Sync messages sent from other devices

williamvds opened the pull request:
When signal-cli syncs messages that have been sent from the user's other
devices, an envelope is produced with the message at
`envelope.syncMessage.sentMessage`.
These are a form of sync messages, whose information is provided under
`envelope.syncMessage`, as compared to normal messages which are under
`envelope.dataMessage`.

Messages are now inserted into the chat history depending on their
timestamp, ensuring that they appear chronologically.
The timestamp of sent messages has been adjusted to be in milliseconds
instead of seconds, as timestamps from signal-cli are in milliseconds.

Accesses to `envelope.dataMessage` have been adjusted to also check
`envelope.syncMessage.sentMessage` so the client can properly handle
synced messages. This is somewhat messy and could be improved in the
future with a mechanism for extracting a 'message' from an envelope,
which would check for both `dataMessage` and `syncMessage.sentMessage`.

Fixes a part of #26

isamert commented on scli:
```diff
@@ -230,14 +259,21 @@ def get_envelope_contact(envelope, signal):
     try:
-        return envelope['dataMessage']['groupInfo']['groupId']
+        if envelope['dataMessage'] is not None:
+            return envelope['dataMessage']['timestamp']
```
I haven't tried the changes but this line should be like this probably, right?
```suggestion
            return envelope['dataMessage']['groupInfo']['groupId']
```

williamvds replied:
Thanks, I've added the `groupInfo` part of it to both normal messages and sync messages

isamert commented on scli:
```diff
@@ -230,14 +259,21 @@ def get_envelope_contact(envelope, signal):
+        if envelope['dataMessage'] is not None:
+            return envelope['dataMessage']['timestamp']
+        if envelope['syncMessage'] is not None:
+            return envelope['syncMessage']['sentMessage']['timestamp']
```
And this:
```suggestion
            return envelope['syncMessage']['sentMessage']['groupId']
```

williamvds commented:
Thanks for the quick review - I've fixed the lack of `groupInfo` in `get_envelope_group_id` and `is_envelope_group_message`.
I also added a cast to `int` on the timestamp to match signal-cli.

isamert commented:
Nice, thanks!
